File: frontend/src/domains/Video.js
import { observable } from 'mobx';
import DomainBase from './DomainBase';

class Video extends DomainBase {
  @observable id;
  @observable titulo;
  @observable descricao;
  @observable duracao;
  @observable tamanhoArquivo;
  @observable infantil;
  @observable dataPublicacao;

  static getDomainAttributes() {
    return ['id', 'titulo', 'descricao', 'duracao', 'tamanhoArquivo', 'infantil', 'dataPublicacao']
  }
}

export default Video;

File: backend/src/main/java/br/gov/ac/tce/licon/entities/Video.java
package br.gov.ac.tce.licon.entities;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;


@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Video extends AbstractAuditableEntity {

	@NotNull
	@Column(name = "TITULO")
	private String titulo;

	@NotNull
	@Column(name = "DESCRICAO")
	private String descricao;

	@NotNull
	@Column(name = "DURACAO")
	private Integer duracao;

	@NotNull
	@Column(name = "TAMANHO_ARQUIVO")
	private Integer tamanhoArquivo;

	@NotNull
	@Column(name = "INFANTIL")
	private Boolean infantil;

	@NotNull
	@Column(name = "DATA_PUBLICACAO")
	private LocalDate dataPublicacao;

}
